File: haskell solutions/problem_6.hs
--program returning the difference between the square sum and the sum of squares of the firs n natural numbers

sum_of_squares::Int -> Int
sum_of_squares n = foldr (\x y -> x*x + y) 0 [1..n]

square_of_sum::Int -> Int
square_of_sum n = 
    let sum = foldr (\x y -> x + y) 0 [1..n]
    in sum^2

difference::Int -> Int
difference n = square_of_sum n - sum_of_squares n

main:: IO()
main = print(difference 100)

File: python solutions/Problems 1-10/problem_6.py
def sum_of_squares(n): #function returning the sum of the squares of the first n natural numbers
    sum = 0
    for i in range(n + 1):
        sum += i*i

    return sum

def square_of_sum(n): #function returning the square of the um of the first n natural numbers
    sum = 0
    for i in range(n + 1):
        sum += i

    return sum*sum

def difference(n): #function returning the difference between the sum of squares and the square of the sum for the first n natural numbers
    return square_of_sum(n) - sum_of_squares(n)

print("This is the difference: ", difference(100))
